---
title: "SARS-CoV-2 and its effects on Africa"
author: "Andrae Ladores"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output:
  pdf_document: default
  html_document:
    df_print: paged
csl: bioinformatics.csl
bibliography: references.bib
params:
  gff_file_path: data/metadata/sars_refgenome_annotation.gff
  vcf_dir_path: data/11_vcf_output_for_R
  sra_runtable_path: data/00_sra_runtable/SraRunTable.txt
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(echo = FALSE,
                      warning = FALSE,
                      message = FALSE,
                      error = TRUE)
```

# Background and Overview
This research analysis is aimed to answer the initial predictions that Africa would be one of the continents that would be heavily affected by the SARS-CoV-2 pandemic [@modellingafrica]. Africa is currently second in the world in terms of total population - containing roughly 1.3 billion people. Africa is also known for their lack of an adequate health care system and they are currently battling and recovering from other epidemics. With this information, it is only expected that Africa would be one of the continents that is deemed vulnerable to SARS-CoV-2 [@africanperspective2020]. 

It has roughly been over a year since the pandemic has affected the entire world. Perversely, Africa is doing extremely well in the fight against the virus. After analyzing relevant COVID-19 datasets, Africa is significantly lower in the number of COVID-19 cases and deaths when compared to other nations.

There are two primary factors why Africa has not been affected by SARS-CoV-2 in the way researchers have predicted. First, the majority of African countries took swift actions to prepare for the inevitable pandemic. Due to the fact that African nations already have prior experience in dealing with viruses or outbreaks, they have a strong nation that understands proper precautions and procedures in order to prevent the spread of the virus [@africacovidpath2020]. Second, while Africa has an enormous population size, it turns out that the majority of the population are adolescents; for example, the average age in Mozambique, Africa wavers around 17 years old. This can make all the difference since the SARS-CoV-2 is known to be fatal towards the older population [@ZHOU20201054]. 

Overall, since Africa seriously prepared for the pandemic, paired with their vast population consisting of young adolescents, Africa was and is able to adequately battle the pandemic and prevent high numbers of cases & mortality rates when compared to other nations.

# Methods
To answer the initial predictions, an adequate COVID-19 dataset is needed; this analysis uses *Our World in Data* or *OWID* COVID-19 dataset. This dataset is compiled using reliable sources, such as *JHU*, *WHO*, *CDC*, and more. It is updated daily and also contains dates, location, continent, and other corresponding information. For this analysis, the dataset was downloaded on May 3rd, 2021. 

Once the *OWID* COVID-19 dataset is acquired, the dataset is then processed in R Studio to organize the data. Once organized using `readr` [@readr], `dplyr` [@dplyr] is used directly after to filter the dataset to only include the African continent and its nations. Then, `ggplot2` [@ggplot2] is applied to create different graphs to assist in the comparison of Africa and outside nations to determine if Africa was or is impacted by SARS-CoV-2.

A Mozambique SRA runtable (SRA Bioproject ID: PRJNA718451) was also used for distinct SNP or variant analysis. It was downloaded on *NCBI* on April 26, 2021. 

# Results and Discussion
## Results
With the figures produced from the *OWID* dataset, Africa contains a mean total reported COVID-19 cases above 80,000 as of recent (Figure 1a). On the other hand, USA has an average total case of well over 30,000,000 (Figure 1b). In a related case, three African countries - Malawi, Mozambique, and Niger - each have a total of under 1,200 COVID-19 related deaths (Figure 2a). With Malawi having the highest out of the three, with a little under 1,200 and Niger having the lowest, containing far below 300 deaths. When compared to three other countries - Italy, United Kingdom, and United States - Italy and United Kingdom fluctuate under 200,000 COVID-19 related deaths, while USA staggers right below 600,000 deaths (Figure 2b).

Despite Africa having an enormous population, pre-existing epidemics, and inadequate health care systems, Africa is successfully fending off COVID-19. Other nations like Italy, United Kingdom, and United States - who have adequate health care systems, smaller total population, and are technologically advanced - have substantial COVID-19 related cases and deaths. 

## Discussion
Africa being minimally affected by SARS-CoV-2 is a great phenomenon, they were able to stay resilient because of two important factors: preparedness and age. 

The majority of African governments efficiently and effectively placed proper COVID-19 procedures - implementing restrictions, closing borders, and spreading proper information to their respective communities [@africanperspective2020]. In addition, Africa's experience with epidemics continues to serve a crucial role. Several African nations have improved their health care systems as a result of prevalent epidemics - malaria, HIV, tuberculosis, and Ebola [@initiativesafrica]. Furthermore, Africans are accustomed to the precautions they need to take to prevent the spread or the infection of the virus - a survey in Mozambique, Africa displayed a compliance rate well above 90% when it came to basic face mask usage, proper cough regimens, and constant hand-washing [@ijerph18031091]. 

In nations like Italy, United Kingdom, and United States, the average population age falters around the 40-50 range (Figure 3a). Compare that to the average age of the African countries - Malawi, Mozambique, and Niger - and there is a clear discrepancy. Malawi, Mozambique, and Niger all have an average population age below 20 (Figure 3b). This crucial aspect greatly contributes to Africa's resilience, since the majority of the population are young and healthy, less COVID-19 related deaths occur [@Diope002699;@NIKOLAI2020112].  

## Rise in numbers
From late December 2020 to early February 2021, Africa experienced a sudden upsurge in new COVID-19 cases (Figure 4a). It isn't clear why this has occurred, but since this sudden spike occurred during prime holiday time, it is most likely due to foreigners and their visits to Africa. These increase of cases is the possible result of high distinct SNPs in places like Mozambique, especially in the S gene - which is directly responsible for binding the virus to the host cells (Figure 4b & 4c). This can be problematic as a mutated S gene can possibly affect the efficacy of the pre-existing COVID-19 vaccines. [@Gilbert2020; @koyama2020variant]


```{r load-packages-and-data}
library("vcfR")
library("ggplot2")
library("dplyr")
library("ggthemes")
library("scales")

# this loads all function files in the code/functions directory
function_paths <- list.files("code/functions", full.names = TRUE)
purrr::walk(function_paths, source)
```

```{r load-owid-data-and-codes}
# read in entire owid dataset
all_owid_data <- readr::read_csv("data/raw_data/owid-covid-data.csv")

# subset owid data, so only Africa is shown
africa_owid_data <- all_owid_data %>%
  dplyr::filter(`continent` == "Africa")

# subset owid data, so only USA is shown
usa_owid_data <- all_owid_data %>%
  dplyr::filter(`location` == "United States")

# subset data, so only Mozambique, Niger, and Malawi is shown.
# Also subset it so only the most recent date is collected
recent_three_african_nations <- all_owid_data %>%
  dplyr::filter(`location` == "Mozambique" | `location` == "Niger" | `location` == "Malawi") %>%
  dplyr::filter(`date` == "2021-05-03")

# subset data, so only USA, UK, and Italy is shown.
# Subset so only the most recent date is collected
recent_three_other_countries <- all_owid_data %>%
  dplyr::filter(`location` == "United States" | `location` == "United Kingdom" | `location` == "Italy") %>%
  dplyr::filter(`date` == "2021-05-03")

# subset data so all of Malawi, Mozambique, and Niger data is shown.
all_three_african_nations <- all_owid_data %>%
  dplyr::filter(`location` == "Mozambique" | `location` == "Niger" | `location` == "Malawi")

# subset data so all of USA, UK, and Italy data is displayed
all_three_other_countries <- all_owid_data %>%
  dplyr::filter(`location` == "United States" | `location` == "United Kingdom" | `location` == "Italy")

# subset owid data, so only Africa is shown
mozambique_owid_data <- all_owid_data %>%
  dplyr::filter(`location` == "Mozambique")

# load color palette 1
cbpalette1 <- c("#999999", "#E69F00", "#56B4E9", "#009E73", "#F0E442", "#0072B2", "#D55E00", "#CC79A7")

# load color palette 2
cbpalette2 <- c("#000000", "#E69F00", "#56B4E9", "#009E73", "#F0E442", "#0072B2", "#D55E00", "#CC79A7")
```

```{r load-vcf-data}
# load in, tidy, and stack all of the VCF files in the target directory
# these are the output of the bash pipeline
stacked_vcfs <- parse_tidy_and_stack_vcfs(
  vcf_dir_path = params$vcf_dir_path)

# load in the gff file with genome annotations for SARS-CoV-2
gff <- read_gff(gff_file_path = params$gff_file_path)

# pull out just the named genes from the annotation file
gene_table <- extract_genes_from_gff(annotation_object = gff)

# combine the stacked vcf table with the gene names and the metadata from
# the sra runtable file
vcf_with_metadata <- add_genes_metadata_to_vcfstack(
  sra_runtable_path = params$sra_runtable_path,
  stacked_vcf = stacked_vcfs,
  cleaned_genes_table = gene_table)
```

# Figures
```{r africa-total-cases}
# graph the mean total mozambique cases over time
africa_total_cases <- africa_owid_data %>%
  group_by(date) %>%
  summarize(mean_total_cases = mean(total_cases)) %>%
  ggplot(aes(x = lubridate::ymd(date),
             y = mean_total_cases)) +
  geom_line(color = "blue") +
  scale_y_continuous(labels = comma) +
  labs(title = ("Mean Total COVID-19 Cases Over Time in Africa"),
       x = "Date",
       y = "Relative Number of Cases")

africa_total_cases
```

**Figure 1a**: The timeline of total COVID-19 cases in Africa.

```{r USA-total-cases}
# graph the total USA cases over time
usa_total_cases <- usa_owid_data %>%
  group_by(date) %>%
  summarize(mean_total_cases = mean(total_cases)) %>%
  ggplot(aes(x = lubridate::ymd(date),
             y = mean_total_cases)) +
  geom_line(color = "green") +
  scale_y_continuous(labels = comma) +
  labs(title = ("Mean Total COVID-19 Cases Over Time in USA"),
       x = "Date",
       y = "Relative Number of Cases")

usa_total_cases
```

**Figure 1b**: The timeline of total COVID-19 cases in USA. 

```{r total-deaths-africa}
# Create line graph that illustrates the total deaths over time in
# three African countries
ggplot(all_three_african_nations, aes(x = date,
                                 y = total_deaths,
                                 colour = location)) +
  geom_line() +
  scale_y_continuous(labels = comma) +
  scale_color_manual("Country", values = cbpalette1) +
  labs(title = "Total Deaths Over Time in Malawi, Mozambique, and Niger",
        x = "Country",
        y = "Number of deaths reported")
```

**Figure 2a**: Total deaths over time in Mozambique, Niger, and Malawi.

```{r total-deaths-other-countries}
# Create line graph that shows total deaths in USA, UK, and Italy
ggplot(all_three_other_countries, aes(x = date,
                                 y = total_deaths,
                                 colour = location)) +
  geom_line() +
  scale_y_continuous(labels = comma) +
  scale_color_manual("Country", values = cbpalette2) +
  labs(title = "Total Deaths Over Time in USA, UK, and Italy",
        x = "Date",
        y = "Number of deaths reported")
```

**Figure 2b**: Total deaths over time in Italy, United Kingdom, and United States. 

```{r median-age-in-three-african-countries}
# create barplot to display the median age over three Africa nations
recent_african_nations_plot <- ggplot(data = recent_three_african_nations,
                                     aes(x = location,
                                         y = median_age,
                                         fill = location)) +
  geom_bar(stat = "identity") +
  scale_fill_manual("Country", values = cbpalette1) +
  labs(title = "Average Age in Mozambique, Niger, and Malawi.",
       x = "Country",
       y = "Median Age")

recent_african_nations_plot
```

**Figure 3a**: Median Age in Mozambique, Niger, and Malawi, Africa. 

```{r median-age-in-three-USA-states}
# create barplot to display the median age over three other nations
recent_other_countries_plot <- ggplot(data = recent_three_other_countries,
                                     aes(x = location,
                                         y = median_age,
                                         fill = location)) +
  geom_bar(stat = "identity") +
  scale_fill_manual("Country", values = cbpalette2) +
  labs(title = "Average Age in United States, United Kingdom, and Italy.",
       x = "Date",
       y = "Median Age")

recent_other_countries_plot
```

**Figure 3b**: Median Age in Italy, United Kingdom, and United State. 

```{r new-cases-mozambique}
# graph newly reported COVID-19 cases in Africa
africa_new_cases <- africa_owid_data %>%
  group_by(date) %>%
  summarize(new_cases_africa = (new_cases)) %>%
  ggplot(aes(x = lubridate::ymd(date),
             y = new_cases_africa)) +
  geom_line(color = "blue") +
  scale_y_continuous(labels = comma) +
  labs(title = ("Newly reported COVID-19 Cases in Africa"),
       x = "Date",
       y = "Number of newly reported COVID-19 cases")

africa_new_cases
```

**Figure 4a**: Number of newly reported COVID-19 cases in Africa

```{r distinct-snps}
# create a plot of unique SNP locations within each gene across all samples
vcf_with_metadata %>%
  filter(!is.na(gene)) %>% # get rid of SNPs not in gene regions
  group_by(gene, pos) %>%
  tally() %>% # this gives a column n for the number of gene by position
  group_by(gene) %>%
  tally() %>% # this collapses that down to the number of unique SNP locations
  ggplot(aes(x = gene,
             y = n)) +
    geom_col() +
    labs(title = "Count of distinct SNPs in Named SARS-CoV-2 Genes",
         x = "Gene Name") +
  theme_few() # get rid of the gray background
```

**Figure 4b**: N and S genes have more unique SNPs in the set of samples analyzed.

# Tables

```{r example-table}
# An example table to show the length of each gene using its start and end
gene_table %>%
  mutate(length = end - start) %>%
  select(gene_name, start, end, length) %>%
  knitr::kable(col.names = c("Gene Name",
                             "Start",
                             "End",
                             "Length"))
```

**Table 4c**: Gene names, locations, and lengths in the SARS-CoV-2 genome. Higher SNP counts in the S and N genes may be related to the larger size of these genes.

# Sources Cited
