---
title: "Why Africa has not been severely affected by the SARS-CoV-2 pandemic"
author: "Andrae Ladores"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output:
  html_document:
    df_print: paged
csl: bioinformatics.csl
bibliography: references.bib
params:
  gff_file_path: data/metadata/sars_refgenome_annotation.gff
  vcf_dir_path: data/11_vcf_output_for_R
  sra_runtable_path: data/00_sra_runtable/SraRunTable.txt
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(echo = FALSE,
                      warning = FALSE,
                      message = FALSE,
                      error = TRUE)
```

# Background and Overview

It was predicted that Africa would be one of the continents that would be heavily affected by the SARS-CoV-2 outbreak. It wasn't surprising as Africa sits at second place in the list of most populated continents, with approximately 1.3 billion (right behind Asia, which stands at 4 billion). Furthermore, Africa is ranked in the third poorest continent in the world, with their lack of adequate health care systems and the fact that Africa is already battling other epidemics: malaria, tuberculosis, and HIV. It wasn't surprising that researchers expected SARS-CoV-2 to have catastrophic damage on Africa. However, that prediction couldn't have been more inaccurate. 

It has been over a year since the pandemic has affected the entire world. Yet, Africa isn't near the top of the most affected nations at all. The primary goal of this research was to directly explore that phenomenon. 

There are two main factors on why this has occurred. Time and the other being age. First, Africa had ample time to prepare for the pandemic - the various governments of Africa took swift action to seriously prepare for the inevitable virus. Second, the majority of the population in African countries are young; the average age in Mozambique wavers around 17 years old. It is known that COVID-19 will severely affect the older population, commonly people that are 65 or older. As a result, since African countries are properly prepared, paired with a large adolescent population, Africa is able to battle the pandemic and prevent high mortality rates, when compared to other nations. 


This is a report on SARS-CoV-2, including some variant analysis [@koyama2020variant].

# Methods

See the set of tutorials on the [vcfR package website](https://knausb.github.io/vcfR_documentation/index.html).

You may also want to use any of a range of different COVID data packages and data sources:

* https://kjhealy.github.io/covdata/
* https://github.com/como-ph/oxcovid19
* https://ropensci.org/blog/2020/10/20/searching-medrxivr-and-biorxiv-preprint-data/
* https://covidtracking.com/data/api
  * `readr::read_csv("https://api.covidtracking.com/v1/states/daily.csv")`

## Subsections are ok too

# Results and Discussion

```{r load-packages-and-functions}
library("vcfR")
library("ggplot2")
library("dplyr")
library("ggthemes")

# this loads all function files in the code/functions directory
function_paths <- list.files("code/functions", full.names = TRUE)
purrr::walk(function_paths, source)
```

```{r load-vcf-data}
# load in, tidy, and stack all of the VCF files in the target directory
# these are the output of the bash pipeline
stacked_vcfs <- parse_tidy_and_stack_vcfs(
  vcf_dir_path = params$vcf_dir_path)

# load in the gff file with genome annotations for SARS-CoV-2
gff <- read_gff(gff_file_path = params$gff_file_path)

# pull out just the named genes from the annotation file
gene_table <- extract_genes_from_gff(annotation_object = gff)

# combine the stacked vcf table with the gene names and the metadata from
# the sra runtable file
vcf_with_metadata <- add_genes_metadata_to_vcfstack(
  sra_runtable_path = params$sra_runtable_path,
  stacked_vcf = stacked_vcfs,
  cleaned_genes_table = gene_table)
```

# Figures

```{r sample-plot}
# create a plot of unique SNP locations within each gene across all samples
vcf_with_metadata %>%
  filter(!is.na(gene)) %>% # get rid of SNPs not in gene regions
  group_by(gene, pos) %>%
  tally() %>% # this gives a column n for the number of gene by position
  group_by(gene) %>%
  tally() %>% # this collapses that down to the number of unique SNP locations
  ggplot(aes(x = gene,
             y = n)) +
    geom_col() +
    labs(title = "Count of distinct SNPs in Named SARS-CoV-2 Genes",
         x = "Gene Name") +
  theme_few() # get rid of the grey background
```

**Figure 1**: N and S genes have more unique SNPs in the set of samples analyzed.

```{r mozambique-total-cases}
# read in entire owid dataset
all_owid_data <- readr::read_csv("data/raw_data/owid-covid-data.csv")

# subset owid data, so only Africa is shown
africa_owid_data <- all_owid_data %>%
  dplyr::filter(`continent` == "Africa")

# graph the mean total mozambique cases over time
africa_total_cases <- africa_owid_data %>%
  group_by(date) %>%
  summarize(mean_total_cases = mean(total_cases)) %>%
  ggplot(aes(x = lubridate::ymd(date),
             y = mean_total_cases)) +
  geom_line(color = "blue") +
  labs(title = ("Relative Mean Total COVID-19 Cases Over-Time in Africa"),
       x = "Date",
       y = "Relative Number of Cases")

africa_total_cases
```

**Figure 2**: The timeline of total COVID-19 cases in Africa.

```{r USA-total-cases}
# subset owid data, so only USA is shown
USA_owid_data <- all_owid_data %>%
  dplyr::filter(`location` == "United States")

# graph the total USA cases over time
USA_total_cases <- USA_owid_data %>%
  group_by(date) %>%
  summarize(mean_total_cases = mean(total_cases)) %>%
  ggplot(aes(x = lubridate::ymd(date),
             y = mean_total_cases)) +
  geom_line(color = "green") +
  labs(title = ("Mean Total COVID-19 Cases Over-Time in USA"),
       x = "Date",
       y = "Relative Number of Cases")

USA_total_cases
```

**Figure 3**: The timeline of total COVID-19 cases in USA. 

```{r median-age-in-three-african-countries}
# load color palette
cbPalette1 <- c("#999999", "#E69F00", "#56B4E9", "#009E73", "#F0E442", "#0072B2", "#D55E00", "#CC79A7")

# subset data, so only Mozambique, Niger, and Malawi is shown.
# Also subset it so only the most recent date is collected
recent_three_african_nations <- all_owid_data %>%
  dplyr::filter(`location` == "Mozambique" | `location` == "Niger" | `location` == "Malawi") %>% 
  dplyr::filter(`date` == "2021-05-03")

recent_three_african_nations_plot <- ggplot(data = recent_three_african_nations,
                                     aes(x = location,
                                         y = median_age,
                                         fill = location)) +
  geom_bar(stat='identity') + 
  scale_fill_manual("Country", values=cbPalette1) +
  labs(title = "Average Age in Mozambique, Niger, and Malawi.",
       x = "Country",
       y = "Median Age")

recent_three_african_nations_plot
```

**Figure 4**: Median Age in Mozambique, Niger, and Malawi, Africa. 

```{r median-age-in-three-USA-states}
# load color palette 
cbPalette2 <- c("#000000", "#E69F00", "#56B4E9", "#009E73", "#F0E442", "#0072B2", "#D55E00", "#CC79A7")

# subset data, so only USA, UK, and Italy is shown.
# Subset so only the most recent date is collected
recent_three_other_countries <- all_owid_data %>%
  dplyr::filter(`location` == "United States" | `location` == "United Kingdom" | `location` == "Italy") %>% 
  dplyr::filter(`date` == "2021-05-03")

recent_three_other_countries_plot <- ggplot(data = recent_three_other_countries,
                                     aes(x = location,
                                         y = median_age,
                                         fill = location)) +
  geom_bar(stat='identity') + 
  scale_fill_manual("Country", values=cbPalette2) +
  labs(title = "Average Age in United States, United Kingdom, and Italy.",
       x = "Country",
       y = "Median Age")

recent_three_other_countries_plot
```

**Figure 5**: Median Age in Italy, United Kingdom, and United State. 

```{r total-deaths-africa}
# subset data so all of Malawi, Mozambique, and Niger data is shown.
all_three_african_nations <- all_owid_data %>%
  dplyr::filter(`location` == "Mozambique" | `location` == "Niger" | `location` == "Malawi")

# Create line graph that illustrates the total deaths over time in
# three African countries 
ggplot(all_three_african_nations, aes(x = date, 
                                 y = total_deaths, 
                                 colour = location)) +
  geom_line() +
  scale_color_manual("Country", values = cbPalette1) +
  labs (title = "Total Deaths Over Time in Malawi, Mozambique, and Niger",
        x = "Country",
        y = "Number of deaths reported")
```

**Figure 6**: Total deaths over time in Mozambique, Niger, and Malawi. 

```{r total-deaths-other-countries}
# subset data so all of USA, UK, and Italy data is displayed
all_three_other_countries <- all_owid_data %>%
  dplyr::filter(`location` == "United States" | `location` == "United Kingdom" | `location` == "Italy") 

# Create line graph that shows total deaths in USA, UK, and Italy
ggplot(all_three_other_countries, aes(x = date, 
                                 y = total_deaths, 
                                 colour = location)) +
  geom_line() +
  scale_color_manual("Country", values = cbPalette2) +
  labs (title = "Total Deaths Over Time in USA, UK, and Italy",
        x = "Country",
        y = "Number of deaths reported")
```

**Figure 6**: Total deaths over time in Italy, United Kingdom, and United States. 


# Tables

```{r example-table}
# An example table to show the length of each gene using its start and end
gene_table %>%
  mutate(length = end - start) %>%
  select(gene_name, start, end, length) %>%
  knitr::kable(col.names = c("Gene Name",
                             "Start",
                             "End",
                             "Length"))
```

**Table 1**: Gene names, locations, and lengths in the SARS-CoV-2 genome. Higher SNP counts in the S and N genes may be related to the larger size of these genes.

# Sources Cited
